master/internal/kubernetes/pod.go

```diff
@@ -71,6 +74,9 @@ func newPod(
 	namespace string,
 	masterIP string,
 	masterPort int32,
+	masterTLSConfig model.TLSClientConfig,
+	loggingTLSConfig model.TLSClientConfig,
+	loggingConfig model.LoggingConfig,
 	podInterface typedV1.PodInterface,
 	configMapInterface typedV1.ConfigMapInterface,
 	resourceRequestQueue *actor.Ref,
```

Comment:
@aaron276h mentioned this but it's a good point, we probably want to add the fluent container to the `containerNames` map below so when we do healthchecks we also terminate on its failures (maybe run this behavior by @vishnu2kmohan, do you want to fail on log failures or keep running without logs?)

Reply:
One more thing to keep in mind here, but unfortunately it's only available is k8s 1.18+ is explicitly labeling it a sidecar container which resolves issues related to startup ordering. 

It is also worth checking some failure scenarios, to make sure we exit properly and don't get stuck in an infinite loop.

Reply:
> when we do healthchecks

Is adding health-checks for the fluent container not in-plan as things stand? It's good practice to enable health-checks for all containers (esp. dependencies), unless there are some good reasons why we don't already have this, today.

> do you want to fail on log failures or keep running without logs?

Ideally we'd fail if we're unable to log. 

@stoksc Are you referring to a startup failure or runtime failure? Can you share a bit more detail on the (expected) failure conditions we need to handle?

Reply:
> when we do healthchecks

I mean when the code does them (we already have them), as is we just never restart and if the container is terminated we assume it has failed, right @aaron276h?

> do you want to fail on log failures or keep running without logs?

Ok, we should do what I'm suggesting then, along with the testing @aaron276h is suggesting.

> One more thing to keep in mind here, but unfortunately it's only available is k8s 1.18+ is explicitly labeling it a sidecar container which resolves issues related to startup ordering.

Interesting, hadn't heard of this, only knew of normal and init containers.

Reply:
Good catch; I added the name to the map for trial containers and tested that the Fluent Bit container failing (either on startup or by being killed later) kills the trial without any issues. Messaging could be better when that happens, but it works.

Did the sidecar thing actually get into Kubernetes? Based on https://github.com/kubernetes/kubernetes/pull/79649 and https://github.com/kubernetes/kubernetes/pull/80744, it looks like it actually got put off, unfortunately.